src/routes/app/wallets/index.tsx

```diff
@@ -1,35 +1,45 @@
 import {
   component$,
+  createContextId,
+  useContextProvider,
   useSignal,
   useStore,
   useVisibleTask$,
+  type Signal,
 } from "@builder.io/qwik";
 import Button from "~/components/Atoms/Buttons/Button";
 import { SelectedWalletDetails } from "~/components/Wallets/Details/SelectedWalletDetails";
-import { type WalletTokensBalances } from "~/interface/walletsTokensBalances/walletsTokensBalances";
-import { chainIdToNetworkName } from "~/utils/chains";
+
 import { ObservedWalletsList } from "~/components/ObservedWalletsList/ObservedWalletsList";
-export {
-  getObservedWallets,
-  ObservedWalletsList,
-} from "~/components/ObservedWalletsList/ObservedWalletsList";
+import { chainIdToNetworkName } from "~/utils/chains";
+export { ObservedWalletsList } from "~/components/ObservedWalletsList/ObservedWalletsList";
 export { useAddWallet, useGetBalanceHistory, useRemoveWallet } from "./server";
 
 import { AddWalletModal } from "./_components/wallets/AddWalletModal";
 import { DeleteModal } from "./_components/wallets/DeleteModal";
-import { balancesLiveStream } from "./server/balancesLiveStream";
-import BoxHeader from "~/components/Molecules/BoxHeader/BoxHeader";
+
 import Input from "~/components/Atoms/Input/Input";
 import Select from "~/components/Atoms/Select/Select";
+import BoxHeader from "~/components/Molecules/BoxHeader/BoxHeader";
+import { balancesLiveStream } from "./server/balancesLiveStream";
+
+export const SelectedWalletDetailsContext = createContextId<Signal<any>>(
+  "selected-wallet-details-context",
+);
+export const SelectedWalletNameContext = createContextId<Signal<string>>(
+  "selected-wallet-name-context",
+);
 
 export default component$(() => {
+  const selectedWalletDetails = useSignal<any | undefined>(undefined);
```

Comment:
could we use stome strict typing istead of any? e.g. use already existing data structure or create a new one?